core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -186,12 +189,23 @@ public PartitionData copy() {
     this.recordCount = toCopy.recordCount;
     this.fileSizeInBytes = toCopy.fileSizeInBytes;
     if (fullCopy) {
-      this.columnSizes = SerializableMap.copyOf(toCopy.columnSizes);
-      this.valueCounts = SerializableMap.copyOf(toCopy.valueCounts);
-      this.nullValueCounts = SerializableMap.copyOf(toCopy.nullValueCounts);
-      this.nanValueCounts = SerializableMap.copyOf(toCopy.nanValueCounts);
-      this.lowerBounds = SerializableByteBufferMap.wrap(SerializableMap.copyOf(toCopy.lowerBounds));
-      this.upperBounds = SerializableByteBufferMap.wrap(SerializableMap.copyOf(toCopy.upperBounds));
+      if (statsToKeep == null || statsToKeep.isEmpty()) {
```

Comment:
to simplify the code, we can probably move the if-else inside the `filterColumnStats` method below. so we can always do `...copyOf(filterColumnsStats(..., columnsToKeepStats)`

Reply:
I did it this way, to save 5 checks, but the code is much nicer if we hide the checks in the method, so refactored